zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -202,21 +210,23 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
                 )
                 .to_word(),
             );
-        });
-        cb.require_equal(
-            "tx nonce equivalence",
-            tx.nonce.expr(),
-            create.caller_nonce(),
-        );
-
-        // 1. Handle contract creation transaction.
-        cb.condition(tx.is_create.expr(), |cb| {
             cb.keccak_table_lookup(
                 create.input_rlc(cb),
                 create.input_length(),
                 caller_nonce_hash_bytes.to_word(),
             );
 
+            cb.keccak_table_lookup(
+                create.code_hash_keccak_rlc(cb),
+                32.expr(),
+                cb.curr.state.code_hash.to_word(),
+            );
+            cb.require_equal_word(
```

Comment:
Do we also need to add this constraint? Cause after creating ContractCreateGadget we checked nonce equivalence, and some other things but not this for example.

Reply:
I think this is not needed.  The `code_hash` in the create gadget seems to be only for the CREATE2 opcode.  The code_hash for creation via BeginTx follows a different pattern (it's hash of tx.calldata).

So here we're using a `create.code_hash` witness from the `ContractCreateGadget` to compare it against another `code_hash`.  But the `create.code_hash` doesn't appear in any constraint inside `ContractCreateGadget` because we're never calling `create.input_rlc` with `IS_CREATE2 = true`.  So the `create.code_hash` will have whichever value the prover assigns, so this equality check is meaningless.